File: algorithm_library/python/math/calc.py
# 切り上げ除算
def ceilDiv(p: int, q: int) -> int:
    return (p + q - 1) // q


# 等差数列の和
# 初項 a1, 項数 n, 公差 d
def cal(a1, n, d):
    if n == 0:
        return 0
    return (n * (2 * a1 + (n - 1) * d)) // 2

File: algorithm_library/cpp-from-py/math/calc.cpp
#include <bits/stdc++.h>
using namespace std;

long long ceilDiv(long long p, long long q) {
    return (p + q - 1) / q;
}

long long max_leq_congruent(long long R, long long r, long long m) {
    r %= m;
    if (r < 0) r += m;
    return R - ((R - r) % m + m) % m;
}

long long min_geq_congruent(long long L, long long r, long long m) {
    assert(m > 0);
    r %= m;
    if (r < 0) r += m;
    return L + ((r - L) % m + m) % m;
}
